### PR TITLE
Calculation Views UI

### PR DESCRIPTION
**Pull request by aptkingston**
## Description
This PR adds the UI for calculation views.

View type is specified upon creation. "Normal" views are called "Table" views.
image

Calculation views are configured using the "Configure calculations" button in the grid toolbar:
image

This opens the calculation editor when you can define which calculations you'd like to add to your view.
image

Example of calculation results:
image

Clarification of functionality:
- Calculation views are completely empty by default. You need to configure some calculations to see data.
- You cannot filter on calculation columns
- You *can* sort on calculation columns
- Calculation views do not support adding, editing or deleting rows
- Calculations can only be defined on numeric columns (number, bigint)
- There is a max of 5 calculations per calculation view
- There is no limit to the number of group bys

### CODE REVIEW

File: packages/builder/src/components/backend/DataTable/buttons/grid/GridViewCalculationButton.svelte [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+        on:click={addCalc}
+        disabled={calculations.length >= 5}
+      >
+        Add calculation
+      </ActionButton>
+    </div>
+    <InfoDisplay
+      icon="Help"
+      quiet
+      body="Calculations only work with numeric columns and a maximum of 5 calculations can be added at once."
```
Comment (shogunpurple) [inline packages/builder/src/components/backend/DataTable/buttons/grid/GridViewCalculationButton.svelte:237]: Is this a hard technical limitation or soft limit?

**Review by shogunpurple — APPROVED** [review]
LGTM! Nice work

File: packages/builder/src/components/backend/DataTable/buttons/grid/GridViewCalculationButton.svelte (hunk 1 above)
Comment (aptkingston) [reply packages/builder/src/components/backend/DataTable/buttons/grid/GridViewCalculationButton.svelte:237]: Soft limit as far as I know. This was just specified in the ticket, but I'm not sure if there's any real reason for it other than to limit the potential performance hit of running these.

**Comment by aptkingston** [comment]
Test failures are random flakes. Going to merge this in and let us use it in QA.

**Pull request merged**